manager/controlapi/service.go

```diff
@@ -163,36 +207,19 @@ func validateTaskSpec(taskSpec api.TaskSpec) error {
 		return grpc.Errorf(codes.InvalidArgument, "TaskSpec: missing runtime")
 	}
 
-	_, ok := taskSpec.GetRuntime().(*api.TaskSpec_Container)
-	if !ok {
+	switch taskSpec.GetRuntime().(type) {
+	case *api.TaskSpec_Container:
+		if err := validateContainerSpec(taskSpec); err != nil {
+			return err
+		}
+	case *api.TaskSpec_Custom:
+		if err := validateCustomSpec(taskSpec.GetCustom()); err != nil {
```

Comment:
Nit: I think it's nicer if functions which match a naming pattern follow a pattern for their arguments/returns to so either all take the type associated with `$1` in `validate(.*)Spec` or all take the generic type and cast themselves. Here you have one of each. Since this is internal and the functions are all pretty near each other it's less important than if it were a public API though.